app/jobs/set_rights_job.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+##
+# Job to update rights for objects
+class SetRightsJob < GenericJob
+  ##
+  # A job that allows a user to update the rights for a list of druids
+  # @param [Integer] bulk_action_id GlobalID for a BulkAction object
+  # @param [Hash] params additional parameters that an Argo job may need
+  def perform(bulk_action_id, params)
+    super
+
+    @new_rights = params[:set_rights][:rights]
+
+    with_bulk_action_log do |log|
+      log.puts("#{Time.current} Starting SetRightsJob for BulkAction #{bulk_action_id}")
+      raise StandardError, 'Must provide rights' if @new_rights.blank?
+
+      update_druid_count
+
+      pids.each do |druid|
+        log.puts("#{Time.current} #{self.class}: Attempting #{druid} (bulk_action.id=#{bulk_action_id})")
+        set_rights(druid, log)
+      end
+      log.puts("#{Time.current} Finished SetRightsJob for BulkAction #{bulk_action_id}")
+    end
+  end
+
+  private
+
+  def set_rights(druid, log)
+    object_client = Dor::Services::Client.object(druid)
+    cocina_object = object_client.find
+    return unless verify_access(cocina_object, log)
+
+    # use dor services client to update the access
+    begin
+      state_service = StateService.new(druid, version: cocina_object.version)
+      raise StandardError, 'Object cannot be modified in its current state.' unless state_service.allows_modification?
+
+      object_client.update(params: cocina_object.new(cocina_update_attributes))
```

Comment:
Have you considered using this code here:
```
form_type = cocina_object.collection? ? CollectionRightsForm : DroRightsForm
form = form_type.new(cocina_object)
form.validate(params)
form.save
```

Reply:
That looks like it will handle a lot of the details of the update, thanks for the pointer! I'm still trying to figure out what `params` should be there...

Reply:
I thought that `form.validate(rights: @new_rights)` would be the thing.  Did you follow it up with `form.save` ?

Reply:
Ah, it did work. I just wasn't passing in the variable the way I thought I was.

Reply:
@jcoyne I'd like to adjust the tests to ensure that the rights down in the [down in the structural] section gets updated too. Is there a good example of how to do that in our rspec?

Reply:
I was wondering if the [a_cocina_object_with_access](https://github.com/sul-dlss/argo/blob/main/spec/support/cocina_matchers.rb#L45-L60) should check access in `structural` too? My tests were passing when those weren't getting updated.

But maybe that could be a separate task? This PR was a bear to rebase because almost every file I touched had a conflict, and it wasn't trivial for me to resolve them.

Reply:
@andrewjbtw I've rebased and pushed some changes to address the file-rights update (thanks for the help @jcoyne). I've tested in my development environment and on QA to make sure it works there. Would you like to take another look before merging?